Доработки по заданию 13 спринта

wrinfotel opened the pull request:
Реализована функциональность:
- Удаление фильмов и пользователей
- Функциональность Отзывы
- Функциональность Общие фильмы
- Добавление режиссёров в фильмы
- Функциональность Лента событий
- Функциональность Поиск
- Функциональность Рекомендации
- Вывод самых популярных фильмов по жанру и годам

wrinfotel committed:
Реализован функционал удаление фильмов и пользователей (#5)

* Реализован функционал удаление фильмов и пользователей

* Иcправления по результатм внутреннего ревью

* Иcправления в тесте в названии функции delete

* Внесены исправления в код по результатам внутреннего ревью

---------

Co-authored-by: Daria Kharkovskaya <dicsi59@gmail.com>

Makar-FP committed:
Add reviews (#8)

* add Reviews

* Remove illegal pattern 'Trailing whitespace'

* Fix

* Fix useful field

* Dot fix

* Fix code duplication

sromashevskaya committed:
Добавлен функционал получения общих фильмов (#10)

* Добавлен функционал получения общих фильмов

* Исправлены замечания после ревью

* Исправлены замечания после ревью

---------

Co-authored-by: svetlana.romashevskaya <lana.romashevskaya@gmail.com>

wrinfotel committed:
Добавление режиссёров в фильмы (#9)

* Добавление режиссёров в фильмы

* рефакторинг сортировки

* добавлен режиссер в ответ по общим фильмам

sromashevskaya committed:
Добавлен функционал получения рекомендованных фильмов (#13)

Co-authored-by: svetlana.romashevskaya <lana.romashevskaya@gmail.com>

Makar-FP committed:
Fix (#11)

dKharkovskaya committed:
Add most populars (#14)

* Добавлены sql запросы по поиску

* Добавлен функционал Вывод самых популярных фильмов по жанру и годам

* Исправление в стилях

* исправление по мерджу

---------

Co-authored-by: Daria Kharkovskaya <dicsi59@gmail.com>

wrinfotel committed:
Функциональность Поиск (#12)

* Функциональность Поиск

* мердж с девелопом и не большие доработки после тестов

* code style fix

wrinfotel committed:
drop tables before run

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/mapper/DirectorMapper.java:
```diff
@@ -0,0 +1,16 @@
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class DirectorMapper {
+
+    public static Director updateDirectorFields(Director oldDirector, Director newDirector) {
```
ммм, это не маппер. Маппер - изменяет объект на другой, например Dto на Entity и наоборот. этот метод `updateDirectorFields()` следует сделать приватным в том классе, в котором он используется

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/mapper/ReviewMapper.java:
```diff
@@ -0,0 +1,27 @@
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ReviewMapper {
+
+    public static Review updateReviewFields(Review oldReview, Review newReview) {
```
смотрите коммент в DirectorMapper

ErkinKasymaliev commented on src/main/java/ru/yandex/practicum/filmorate/model/feed/Feed.java:
```diff
@@ -0,0 +1,20 @@
+@Data
+@Builder
+public class Feed {
+    Long timestamp;
```
не забываем про модификаторы доступа

ErkinKasymaliev reviewed (changes requested):
привет, оставил комменты, но есть один момент: где в запросах вы указываете схему public, где то нет. Оставьте один вариант -- лучше уберите схему

ErkinKasymaliev reviewed (changes requested):
привет, оставил комменты, но есть один момент: где в запросах вы указываете схему public, где то нет. Оставьте один вариант -- лучше уберите схему

wrinfotel committed:
доработки после ревью
- убрали схему из запросов
- добавили модификаторы доступа
- перенесли изменение полей при обновлении в сервисы

ErkinKasymaliev reviewed (approved):
работа принята
